### ISSUE TITLE
Panic with class highlighter

### ISSUE THREAD

**Issue by Keats** [trishume/syntect#336]
To reproduce:

1. Update sublime package subrepo to `207cf862`
2. Run the following test:

```
    #[test]
    fn test_classed_html_generator_panics() {
        let current_code = "{\n    \"headers\": [\"Number\", \"Title\"],\n    \"records\": [\n        [\"1\", \"Gutenberg\"],\n        [\"2\", \"Printing\"]\n    ],\n}\n";
        let syntax_set = SyntaxSet::load_defaults_newlines();
        let syntax = syntax_set.find_syntax_by_token("json").unwrap();
        println!("{:?}", syntax);
        let mut html_generator = ClassedHTMLGenerator::new_with_class_style(&syntax, &syntax_set, ClassStyle::Spaced);
        for line in LinesWithEndings::from(current_code) {
            html_generator.parse_html_for_line_which_includes_newline(&line);
        }
        let html = html_generator.finalize();
        assert_eq!(html, "<span class=\"source r\">x <span class=\"keyword operator arithmetic r\">+</span> y\n</span>");
    }
```
It will panic with:

```
thread 'html::tests::test_classed_html_generator_panics' panicked at 'tried to restore cleared scopes, but none were cleared', src/parsing/scope.rs:450:29
```

Making `src/parsing/scope.rs:450:29` a no-op (`()` instead of panicking) fixes the issue and it seems highlighted correctly (?).

**Comment by keith-hall** [comment]
Thanks for reporting. I believe my comment/diagnosis at https://github.com/trishume/syntect/issues/326#issuecomment-792057860 applies here.

**Comment by Keats** [comment]
PR: https://github.com/trishume/syntect/pull/337

**Comment by Enselic** [comment]
Closed by the merged PR #337
